### PR TITLE
call with all but one 64th of the maximum allowed amount of gas

### PR DESCRIPTION
**Pull request by ayrat555**
From EIP150:
```
If a call asks for more gas than the maximum allowed amount (i.e. the total amount of gas remaining in
the parent after subtracting the gas cost of the call and memory expansion), do not return an OOG 
error; instead, if a call asks for more gas than all but one 64th of the maximum allowed amount, call
with all but one 64th of the maximum allowed amount of gas (this is equivalent to a version of EIP-901
 plus EIP-1142). CREATE only provides all but one 64th of the parent gas to the child call.
```

### CODE REVIEW

File: apps/evm/lib/evm/gas.ex [hunk 1]
```diff
@@ -129,7 +139,48 @@ defmodule EVM.Gas do
     operation_cost = operation_cost(operation.sym, inputs, machine_state, exec_env)
     memory_cost = memory_cost(operation.sym, inputs, machine_state)
 
-    memory_cost + operation_cost
+    gas_cost = memory_cost + operation_cost
+
+    if operation.sym in @call_operations && machine_state.gas < gas_cost do
```
Comment (masonforest) [inline apps/evm/lib/evm/gas.ex]: could we extract this into a private method?

File: apps/evm/lib/evm/gas.ex [hunk 2]
```diff
@@ -129,7 +139,48 @@ defmodule EVM.Gas do
...
+        else
+          gas_cost
+        end
+      end
+    else
+      gas_cost
+    end
+  end
+
+  def cost_wth_state_update(machine_state, exec_env) do
```
Comment (masonforest) [inline apps/evm/lib/evm/gas.ex]: wth -> with

File: apps/evm/lib/evm/gas.ex [hunk 3]
```diff
@@ -129,7 +139,48 @@ defmodule EVM.Gas do
...
+
+    if operation.sym in @call_operations && machine_state.gas < gas_cost do
+      if Configuration.fail_nested_operation_lack_of_gas?(exec_env.config) do
+        gas_cost
+      else
+        stack_exec_gas = List.first(inputs)
+        call_cost_without_exec_gas = gas_cost - stack_exec_gas
+        remaining_gas = machine_state.gas - call_cost_without_exec_gas
+
+        if remaining_gas > 0 do
```
[the remark this answers is not in the document]
Comment (ayrat555) [reply apps/evm/lib/evm/gas.ex]: @germsvel @masonforest  I just extracted it. Can you take a look if it's ok?

[the remark this answers is not in the document]
Comment (ayrat555) [reply apps/evm/lib/evm/gas.ex]: I will tests for this logic when I'll fix common tests

**Pull request merged**